src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -434,5 +403,14 @@
              flashlight-icon])]]))))
 
 (defn view
-  [props]
-  [:f> f-view props])
+  [_props]
+  (let [qr-code-succeed? (reagent/atom false)]
+    (navigation.util/create-class-and-bind
+     "sign-in-intro"
+     {:component-did-appear (fn set-qr-code-failed [_this]
+                              (reset! qr-code-succeed? false))}
+     (fn [props]
+       (let [new-pops (assoc props
+                             :qr-code-succeed?      @qr-code-succeed?
+                             :set-qr-code-succeeded #(reset! qr-code-succeed? true))]
+         [:f> f-view new-pops])))))
```

Comment:
This is the important part of the solution:

I am using the new helper function I mentioned in this PR's description. as I said before, I'd like it to be a macro.

It will reset the atom `qr-code-suceeed?` to `false` each time this screen is shown to the user, so we can re-read the code again.

In react-native-navigation, the screen components are not unmounted when we navigate to another screen, it means:
- the Cleanup function in `use-effect`
- the `finally` statement in `with-let`
- the componentWillUnmount() method

will not be called when we navigate to another screen in the same stack, they will be called when we detach the screens from navigation hierarchy, but in cases like this (where we want to go back to a screen and we expect its state is reset) we need to use the `componentDidAppear` method given by `react-native-navigation`.

So, the atom `qr-code-succeeed?` is not recreated when we hit the "try again" button to navigate back.

I think now that this has been noticed, we will start using these extended lifecycle methods more often, so I decided to create a helper function.